community/cloud-foundation/templates/gcs_bucket/gcs_bucket.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2018 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Create a storage bucket resource. """
+import string
+import random
+
+
+def generate_config(context):
+    """
+    Entry point for the deployment resources
+    if autoGenerateName is True then generate name
+    by appending provided name with a random_string.
+    """
+
+    resources = []
+    random_string_len = 10
+
+    project_id = context.env['project']
+    bucket_name = context.properties.get('name') or context.env['name']
+
+    autogenerate_name = context.properties.get('autoGenerateName')

```

Comment:
Please remove the 'autoGenerateName' functionality, this is a bad practice because every update will delete and create a new bucket. Random has to come from outside the DM Temaplete ( like a property.)

Reply:
Good point. Updated in `6adcb06`